spec/models/spree/gateway/stripe_sca_spec.rb

```diff
@@ -3,80 +3,117 @@
+                                     },
+                                   })
+    end
+    let!(:payment_intent) do
+      Stripe::PaymentIntent.create({
+                                     amount: 1000, # given in AUD cents
+                                     currency: 'aud', # AUD to match order currency
+                                     payment_method: pm_card,
+                                     payment_method_types: ['card'],
+                                     capture_method: 'manual',
+                                   })
+    end
+
+    # Stripe acepts amounts as positive integers representing how much to charge
+    # in the smallest currency unit
+    let(:capture_amount) { order.total.to_i * 10 }
+    let(:response) { subject.purchase(capture_amount, credit_card, gateway_options) }
+
+    before do
+      # confirms the payment
+      Stripe::PaymentIntent.confirm(payment_intent.id)
+    end
+
+    it "completes the purchase" do
+      payment
 
-      response = subject.purchase(order.total, credit_card, gateway_options)
+      expect {
+        response.to
+        change(
+          Stripe::PaymentIntent.retrieve(payment_intent.id).status
+        ).from("requires_capture").to("suceeded")
+      }
 
       expect(response.success?).to eq true
     end
 
     it "provides an error message to help developer debug" do
-      stub_request(:get, "https://api.stripe.com/v1/payment_intents/12345").
-        to_return(status: 200, body: capture_successful)
+      response_error = subject.purchase(capture_amount, credit_card, gateway_options)
 
-      response = subject.purchase(order.total, credit_card, gateway_options)
+      expect(response_error.success?).to eq false
+      expect(response_error.message).to eq "No pending payments"
+    end
+  end
 
-      expect(response.success?).to eq false
-      expect(response.message).to eq "Invalid payment state: succeeded"
+  context "#error message", :vcr, :stripe_version do
+    before { Stripe.api_key = secret }
+
+    let!(:pm_card) do
+      Stripe::PaymentMethod.create({
+                                     type: 'card',
+                                     card: {
+                                       number: '4242424242424242',
+                                       exp_month: 12,
+                                       exp_year: year_valid,
+                                       cvc: '314',
+                                     },
+                                   })
+    end
+    let!(:payment_intent) do
+      Stripe::PaymentIntent.create({
+                                     amount: 1000, # given in AUD cents
+                                     currency: 'aud', # AUD to match order currency
+                                     payment_method: pm_card,
+                                     payment_method_types: ['card'],
+                                     capture_method: 'manual',
+                                   })
     end
```

Comment:
These two `let`s are the same as above, so could be moved up one block and shared without the need to repeat.

And in that case, I think it would be fine to also embed `Time.zone.now.year.next` without the need for another `let`.

Reply:
I was having issues with DRYing this. I had tried before to have a private method for it, which did not work, and I was not being able to implement it now as a shared example.

In fact, it's much simpler: all we need to do is call this with `let` instead of `let!`. This way, it will run the code again, and create separate payment intents.